### ISSUE TITLE
Image Search Inside function unintended return

### ISSUE THREAD

**Issue by 90301** [Pulover/PuloversMacroCreator#110]
I'm not sure if this is a nesting issue or what, but when I have a function that attempts an image search it always returns 1, but I can never get to the code inside the if statement.

Here's my function:
```
CheckMission()
{
    MsgBox, 0, , starting function
    CoordMode, Pixel, Window
    ImageSearch, FoundX1, FoundY1, 13, 30, 1024, 793, *15 C:\Img1.png
    If ErrorLevel = 0
    {
        MsgBox, 0, , Found Mission 1 %FoundX% %FoundY%
        return 1
    }
    Else
    {
        MsgBox, 0, , Nope 1
    }
    MsgBox, 0, , Check 2
    CoordMode, Pixel, Window
    ImageSearch, FoundX2, FoundY2, 13, 30, 1024, 793, *15 C:\Img2.png
    If ErrorLevel = 0
    {
        MsgBox, 0, , Found Mission 2 %FoundX2% %FoundY2%
        return 1
    }
    MsgBox, 0, , Not Found
    return 0
}
```

This is calling my function:

```
WinActivate, myWindow
Sleep, 333
MissionFound := CheckMission()
MsgBox, 0, , --- %MissionFound% %true%
If MissionFound == 1
{
    MsgBox, 0, , Mission Found %MissionFound% %true%
    Break
}
Else
{
    MsgBox, 16, , %MissionFound%
}

```

And what happens is:
* I get the message box `starting function`
* Then I get the message box `--- 1 1`
* And finally the message box `Mission Found 1 1`

I can't get it to get past the first image search.

----

Anyway, awesome program you have built. I'm going to try a workaround by not making this a function, and instead just using a macro and trying to use global variables, but structurally, I would like to be able to use this as a function.

**Comment by Pulover** [comment]
Thanks for the report. I tested your scenario and confirmed that there is a bug in function return, so I'm going to work on fixing it.
It also caught my attention that PMC is still exporting compared variables in old fashion (without parenthesis) so I'm updating it to export as expression.

**Comment by Pulover** [comment]
Fixed in v5.2.3.
